src/App.tsx

```diff
@@ -1,31 +1,57 @@
+import { useState } from 'react';
 import './App.css';
 import Button from './components/Button/Button';
 import Input from './components/Input/Input';
+import RepoCard from './components/RepoCard/RepoCard';
+import { GithubRepo, fetchGithubRepos } from './GithubAPI';
 
 function App() {
+  const [showRepos, setShowRepos] = useState<GithubRepo[]>([]);
+
+  const getRepos = async (e: any) => {
+    e.preventDefault();
+    try {
+      let githubResponse = await fetchGithubRepos(
+        (e.currentTarget as HTMLFormElement).username.value
+      );
+      if (githubResponse.ok) {
+        setShowRepos(githubResponse.repos);
+      } else {
+        alert(githubResponse.message);
+      }
+    } catch (err) {
+      console.log(err);
+      alert('Something went wrong. Refresh and try again!');
+    }
+    e.target.reset();
+  };
+
   return (
     <div className="app">
-      <main>
-        <header className="text-center">
-          <h1> Search repos </h1>
-        </header>
-        <form className="flex justify-center">
-          <Input
-            placeholder="username"
-            name="username"
-            type={'search'}
-            className="rounded-r-none"
-          />
-          <Button
-            type="submit"
-            variant="outline"
-            color="black"
-            className="bg-white rounded-l-none"
-          >
-            Submit
-          </Button>
-        </form>
-      </main>
+      <header className="text-center">
+        <h1> Search repos </h1>
+      </header>
+      <form onSubmit={getRepos} className="flex justify-center">
+        <Input
+          placeholder="username"
+          name="username"
+          type={'search'}
+          className="rounded-r-none"
+        />
+        <Button
+          type="submit"
+          variant="outline"
+          color="black"
+          className="bg-white rounded-l-none"
+        >
+          Submit
+        </Button>
+      </form>
+      <div className="grid grid-cols-3 mt-4 gap-4 text-base text-start">
```

Comment:
We should have a minimal "loading" state. For now a simple "Loading" button can do, before we have SkeletonCards